esmvaltool/diag_scripts/ocean/diagnostic_maps_multimodel.py

```diff
@@ -0,0 +1,316 @@
+                hascbar = True
+
+            if ii == 0:
+                thename = thename + ' (' + varname + ') [' + varunit + ']'
+
+            if (hasobs) & (ii > 0):
+                cube = diff_cubes[ii - 1]
+                nspace = np.linspace(
+                    diff_range[0], diff_range[1], clevels, endpoint=True)
+                cmap = 'RdBu_r'
+                thename = thename
+
+            axs = plt.subplot(gsc[xx, yy], projection=proj)
+            add_map_plot(
+                axs,
+                cube,
+                nspace,
+                yy,
+                cmap=cmap,
+                title=thename,
+                extend=extend,
+                hascbar=hascbar)
+
+            # next row & column indexes
+            xx = xx + 1
+            if xx == layout[0]:
+                xx = 1 if hasobs else 0
+                yy = yy + 1
+
+        # Adjust subplots size & position
+        plt.subplots_adjust(
+            top=0.92,
+            bottom=0.08,
+            left=0.05,
+            right=0.95,
+            hspace=0.15,
+            wspace=0.15)
+
+        # Vertically detach OBS plot and center
+        if hasobs:
+            axs = fig.axes
+            box = axs[0].get_position()
+            shift = box.y0 * 0.05
+            box.y0 = box.y0 + shift
+            box.y1 = box.y1 + shift
+            shift = box.x1 - box.x0
+            box.x0 = 0.5 - shift * 0.5
+            box.x1 = box.x0 + shift
+            axs[0].set_position(box)
+
+        # Determine image filename:
+        fn_list = ['multimodel_vs', obsname, varname, str(layer), 'maps']
+        path = diagtools.folder(cfg['plot_dir']) + '_'.join(fn_list)
+        path = path.replace(' ', '') + image_extention
+
+        # Saving files:
+        if cfg['write_plots']:
+            logger.info('Saving plots to %s', path)
+            plt.savefig(path, dpi=200)
+
+        plt.close()
+
+
+def main(cfg):
+    """
+    Load the config file, and send it to the plot maker.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionairy, passed by ESMValTool.
+
+    """
+    for index, metadata_filename in enumerate(cfg['input_files']):
+        logger.info(
+            'metadata filename:\t%s, %s',
+            index,
+            metadata_filename,
+        )
+        metadatas = diagtools.get_input_files(cfg, index=index)
+
+        model_type = 'observational_dataset'
+        logger.debug(
+            'model_type: %s, %s',
+            index,
+            model_type,
+        )
+        logger.debug(
+            'metadatas:  %s, %s',
+            index,
+            metadatas,
+        )
+        obs_filename = diagtools.match_model_to_key('observational_dataset',
+                                                    cfg[model_type], metadatas)
+
+        if not os.path.exists(obs_filename):
+            logger.info('OBS file not found %s', obs_filename)
+            obs_filename = ''
+
```

Comment:
`None` and therefore can be optional in the other func upstream

Reply:
changed accordingly.